src/App.tsx

```diff
@@ -31,62 +31,98 @@ import ProtectedRoute from './components/ProtectedRoute/ProtectedRoute';
-    authCheckLoginState();
-  }, [token, authCheckLoginState])
-
-  const requester_routes = (
+const RequesterRoutes = () => {
+  return (
     <Switch>
-      <ProtectedRoute exact path='/my-requests' component={RequestList} />
-      <ProtectedRoute exact path='/my-requests/create-request' component={CreateRequest} />
-      <ProtectedRoute exact path='/my-requests/:requestId' component={RequestDetail} />
+      <ProtectedRoute exact path="/my-requests" component={RequestList} />
+      <ProtectedRoute
+        exact
+        path="/my-requests/create-request"
+        component={CreateRequest}
+      />
+      <ProtectedRoute
+        exact
+        path="/my-requests/:requestId"
+        component={RequestDetail}
+      />
     </Switch>
   );
-  const volunteer_routes = (
-    <Switch>
-      <ProtectedRoute exact path='/volunteer' component={VolunteerList} />
-      <ProtectedRoute exact path='/volunteer/my-support' component={MyVolunteerList} />
-      <ProtectedRoute exact path='/volunteer/my-support/:volunteerId' component={MyVolunteerDetail} />
-      <Redirect from='/volunteer/signup' to='/volunteer' />
-      <ProtectedRoute exact path='/volunteer/:requestId' component={VolunteerDetail} />
-      <Redirect exact from='/profile' to='/profile/me/' />
-      <Route exact path='/profile/:userId/' component={UserProfile} />
-    </Switch>
-  )
-  const public_routes = (
+};
+
+const VolunteerRoutes = () => {
+  return (
     <Switch>
-      <Route exact path='/' component={Home} />
-      <Route exact path='/get-help' component={GetHelp} />
-      <Route exact path='/get-involved' component={GetInvolved} />
-      <Route exact path='/volunteer/signup' component={SignUp} />
-      <Redirect from="/admin" to="/admin/" />
+      <ProtectedRoute exact path="/volunteer" component={VolunteerList} />
+      <ProtectedRoute
+        exact
+        path="/volunteer/my-support"
+        component={MyVolunteerList}
+      />
+      <ProtectedRoute
+        exact
+        path="/volunteer/my-support/:volunteerId"
+        component={MyVolunteerDetail}
+      />
+      <Redirect from="/volunteer/signup" to="/volunteer" />
+      <ProtectedRoute
+        exact
+        path="/volunteer/:requestId"
+        component={VolunteerDetail}
+      />
+      <Redirect exact from="/profile" to="/profile/me/" />
+      <Route exact path="/profile/:userId/" component={UserProfile} />
     </Switch>
   );
+};
+
+const App = ({
+  isAuthenticated,
+  authCheckLoginState,
+  is_requester,
+  is_volunteer,
+}) => {
+  const token = localStorage.getItem('access');
+
+  useEffect(() => {
+    authCheckLoginState();
+  }, [token, authCheckLoginState]);
 
   return (
     <Layout>
-      {public_routes}
+      <Switch>
+        <Route exact path="/" component={Home} />
+        <Route exact path="/get-help" component={GetHelp} />
+        <Route exact path="/get-involved" component={GetInvolved} />
+        <Route exact path="/volunteer/signup" component={SignUp} />
```

Comment:
we probably should change volunteer/signup route to avoid `<ProtectedRoute exact path='/volunteer/:requestId' component={VolunteerDetail} />`

```suggestion
        <Route exact path="/volunteer-signup" component={SignUp} />
```